### PR TITLE
[menu] RadioItem component

### LINKED ISSUE #53: [Menu] Create the radio menu item
Create a component that acts as a selectable menu item. It should follow the spec for the ARIA `menuitemradio` role (https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/menuitemradio_role).

### PR DESCRIPTION
**Pull request by michaldudak**
Created the MenuItemRadioItem and related components.

Closes #53

### CODE REVIEW

File: packages/mui-base/src/Menu/RadioItem/MenuRadioItem.tsx [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+import { useComponentRenderer } from '../../utils/useComponentRenderer';
+import { CustomStyleHookMapping } from '../../utils/getStyleHookProps';
+import { useId } from '../../utils/useId';
+import type { BaseUIComponentProps, GenericHTMLProps } from '../../utils/types';
+import { useForkRef } from '../../utils/useForkRef';
+import { useMenuRadioGroupContext } from '../RadioGroup/MenuRadioGroupContext';
+import { MenuRadioItemContext } from './MenuRadioItemContext';
+
+const customStyleHookMapping: CustomStyleHookMapping<MenuRadioItem.OwnerState> = {
+  checked: (value: boolean) => ({ 'data-state': value ? 'checked' : 'unchecked' }),
```
Comment (atomiks) [inline packages/mui-base/src/Menu/RadioItem/MenuRadioItem.tsx:16]: `data-radioitem`? Side note: `data-radio-item` instead? Checkbox had `data-checkboxitem`

File: packages/mui-base/src/Menu/Positioner/useMenuPositioner.ts [hunk 2]
```diff
@@ -43,6 +43,7 @@ export function useMenuPositioner(
           zIndex: 2147483647, // max z-index
         },
         'aria-hidden': !open || undefined,
+        inert: !open || undefined ? '' : undefined,
```
Comment (atomiks) [inline packages/mui-base/src/Menu/Positioner/useMenuPositioner.ts:46]: What is `|| undefined` for?

Comment (michaldudak) [reply packages/mui-base/src/Menu/Positioner/useMenuPositioner.ts:46]: 😲 copy-pasta...

File: packages/mui-base/src/Menu/RadioItem/MenuRadioItem.tsx (hunk 1 above)
Comment (michaldudak) [reply packages/mui-base/src/Menu/RadioItem/MenuRadioItem.tsx:16]: Both `radio-item` and `radioitem` are fine. I'd be consistent and have no hyphens (this also aligns with ARIA roles)

File: packages/mui-base/src/Menu/RadioGroup/MenuRadioGroup.tsx [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+    name: 'MenuRadioGroup',
+  });
+
+  const setValue = React.useCallback(
+    (newValue: any, event: Event) => {
+      setValueUnwrapped(newValue);
+      onValueChange?.(newValue, event);
+    },
+    [onValueChange, setValueUnwrapped],
+  );
```
Comment (atomiks) [inline packages/mui-base/src/Menu/RadioGroup/MenuRadioGroup.tsx:28]: `useEventCallback`

**Comment by colmtuite** [comment]
@michaldudak I was surprised that the Menu doesn't close when you select a radio item using Enter. Radix closes the menu whether you select a radio/checkbox item via click, space, or Enter. RA closes on Enter only.

Do you think we should keep the Menu open always? Or close when you select a radio item using Enter?

**Comment by michaldudak** [comment]
Yeah, it should close after Enter is pressed. I'll fix it.

**Comment by michaldudak** [comment]
I made the change. I won't update the CheckboxItem, as it's the same code. Once RadioItem gets merged, the CheckboxItem will work automatically.

**Review by atomiks — APPROVED** [review]

**Review by colmtuite — APPROVED** [review]

**Pull request merged**